.github/dependabot.yml

```diff
@@ -4,26 +4,42 @@
 version: 2
 updates:
   # This is to update requirements.txt files in the guided-demos, and e2e directories.
-  # The group configuration option is used to group updates for consistency across related directories.
   - package-ecosystem: "pip"
     directories:
       - "**/demo-notebooks/guided-demos*"
       - "/tests/e2e"
     schedule:
-      interval: "weekly"
-    groups:
-      requirements.txt:
-        patterns:
-          - "*"
-    open-pull-requests-limit: 10
+      interval: "daily"
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-patch"]
+    open-pull-requests-limit: 4
     labels:
+      - "dependabot"
       - "test-guided-notebooks"
 
   # pip means poetry in this case, this keeps poetry.lock up to date with constraints in pyproject.toml.
   - package-ecosystem: "pip"
     directory: "/"
     schedule:
-      interval: "weekly"
-    open-pull-requests-limit: 10
+      interval: "daily"
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-patch"]
```

Comment:
Why are we ignoring patch updates?

Reply:
Good question, this is to avoid creating a larger number of PRs and use of resources for minor patches. A PR will still be created for `x.y` releases. In scenarios where we would get a critical/high CVEs and is resolved in a patch we could manually address them. If it does become an ongoing issue and requires regular manual updates we can definitely remove this parameter.